src/platforms/posix/drivers/df_mpu6050_wrapper/df_mpu6050_wrapper.cpp

```diff
@@ -0,0 +1,712 @@
+
+		if (res != OK) {
+			PX4_ERR("Could not access param %s", str);
+		}
+
+		(void)sprintf(str, "CAL_ACC%u_ZSCALE", i);
+		res = param_get(param_find(str), &_accel_calibration.z_scale);
+
+		if (res != OK) {
+			PX4_ERR("Could not access param %s", str);
+		}
+
+		(void)sprintf(str, "CAL_ACC%u_XOFF", i);
+		res = param_get(param_find(str), &_accel_calibration.x_offset);
+
+		if (res != OK) {
+			PX4_ERR("Could not access param %s", str);
+		}
+
+		(void)sprintf(str, "CAL_ACC%u_YOFF", i);
+		res = param_get(param_find(str), &_accel_calibration.y_offset);
+
+		if (res != OK) {
+			PX4_ERR("Could not access param %s", str);
+		}
+
+		(void)sprintf(str, "CAL_ACC%u_ZOFF", i);
+		res = param_get(param_find(str), &_accel_calibration.z_offset);
+
+		if (res != OK) {
+			PX4_ERR("Could not access param %s", str);
+		}
+
+		// We got calibration values, let's exit.
+		return;
+	}
+
+	// Set sane default calibration values
+	_accel_calibration.x_scale = 1.0f;
+	_accel_calibration.y_scale = 1.0f;
+	_accel_calibration.z_scale = 1.0f;
+	_accel_calibration.x_offset = 0.0f;
+	_accel_calibration.y_offset = 0.0f;
+	_accel_calibration.z_offset = 0.0f;
+}
+
+int DfMPU6050Wrapper::_publish(struct imu_sensor_data &data)
+{
+	/* Check if calibration values are still up-to-date. */
+	bool updated;
+	orb_check(_param_update_sub, &updated);
+
+	if (updated) {
+		parameter_update_s parameter_update;
+		orb_copy(ORB_ID(parameter_update), _param_update_sub, &parameter_update);
+
+		_update_accel_calibration();
+		_update_gyro_calibration();
+	}
+
+	math::Vector<3> vec_integrated_unused;
+	uint64_t integral_dt_unused;
+
+	math::Vector<3> accel_val((data.accel_m_s2_x - _accel_calibration.x_offset) * _accel_calibration.x_scale,
+				  (data.accel_m_s2_y - _accel_calibration.y_offset) * _accel_calibration.y_scale,
+				  (data.accel_m_s2_z - _accel_calibration.z_offset) * _accel_calibration.z_scale);
+
+	// apply sensor rotation on the accel measurement
+	accel_val = _rotation_matrix * accel_val;
+
+	_accel_int.put_with_interval(data.fifo_sample_interval_us,
+				     accel_val,
+				     vec_integrated_unused,
+				     integral_dt_unused);
+
+	math::Vector<3> gyro_val((data.gyro_rad_s_x - _gyro_calibration.x_offset) * _gyro_calibration.x_scale,
+				 (data.gyro_rad_s_y - _gyro_calibration.y_offset) * _gyro_calibration.y_scale,
+				 (data.gyro_rad_s_z - _gyro_calibration.z_offset) * _gyro_calibration.z_scale);
+
+	// apply sensor rotation on the gyro measurement
+	gyro_val = _rotation_matrix * gyro_val;
+
+	_gyro_int.put_with_interval(data.fifo_sample_interval_us,
+				    gyro_val,
+				    vec_integrated_unused,
+				    integral_dt_unused);
+
+	// If we are not receiving the last sample from the FIFO buffer yet, let's stop here
+	// and wait for more packets.
+	if (!data.is_last_fifo_sample) {
+		return 0;
+	}
+
+	// The driver empties the FIFO buffer at 1kHz, however we only need to publish at 250Hz.
+	// Therefore, only publish every forth time.
```

Comment:
The sampling on the device itself happens with 1kHz and written into a FIFO queue and the driver's measurement interval is 1000 us. So far I didn't have any issues with FIFO overflows or corrupted data and the measurements are reasonable.
Should I test something else?